src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -272,7 +272,7 @@ ErrorIfModifyQueryNotSupported(Query *queryTree)
 				continue;
 			}
 
-			if (!IsA(targetEntry->expr, Const))
+			if (contain_mutable_functions((Node *) targetEntry->expr))
```

Comment:
On line 335 there is a check for IsA(setTargetEntry->expr, Var). Do you or @onderkalaci know why there is an explicit check for that. Is there a case that it not covered by contain_mutable_functions that we can support, or is this something specific to upsert?